plugins/auth-backend/src/providers/github/provider.test.ts

```diff
@@ -0,0 +1,195 @@
+            accessToken: '19xasczxcm9n7gacn9jdgm19me',
+            expiresInSeconds: undefined,
+            idToken: undefined,
+            scope: 'read:scope',
+          },
+          profile: {
+            displayName: 'Jimmy Markum',
+            picture:
+              'https://a1cf74336522e87f135f-2f21ace9a6cf0052456644b80fa06d4f.ssl.cf2.rackcdn.com/images/characters_opt/p-mystic-river-sean-penn.jpg',
+          },
+        },
+      },
+      {
+        testCase:
+          'when "displayName" is missing, it should be able to create the profile and map "displayName" with "username"',
+        arguments: {
+          accessToken: '19xasczxcm9n7gacn9jdgm19me',
+          rawProfile: {
+            id: 'uid-123',
+            username: 'jimmymarkum',
+            provider: 'github',
+            displayName: null,
+            emails: null,
+            photos: [
+              {
+                value:
+                  'https://a1cf74336522e87f135f-2f21ace9a6cf0052456644b80fa06d4f.ssl.cf2.rackcdn.com/images/characters_opt/p-mystic-river-sean-penn.jpg',
+              },
+            ],
+          },
+          params: {
+            scope: 'read:scope',
+          },
+        },
+        expect: {
+          backstageIdentity: {
+            id: 'jimmymarkum',
+          },
+          providerInfo: {
+            accessToken: '19xasczxcm9n7gacn9jdgm19me',
+            expiresInSeconds: undefined,
+            idToken: undefined,
+            scope: 'read:scope',
+          },
+          profile: {
+            displayName: 'jimmymarkum',
+            picture:
+              'https://a1cf74336522e87f135f-2f21ace9a6cf0052456644b80fa06d4f.ssl.cf2.rackcdn.com/images/characters_opt/p-mystic-river-sean-penn.jpg',
+          },
+        },
+      },
+      {
+        testCase:
+          'when "photos" is missing, it should be able to create the profile without it',
+        arguments: {
+          accessToken:
+            'ajakljsdoiahoawxbrouawucmbawe.awkxjemaneasdxwe.sodijxqeqwexeqwxe',
+          rawProfile: {
+            id: 'ipd12039',
+            username: 'daveboyle',
+            provider: 'gitlab',
+            displayName: 'Dave Boyle',
+            emails: [
+              {
+                value: 'daveboyle@gitlab.org',
+              },
+            ],
+          },
+          params: {
+            scope: 'read:user',
+          },
+        },
+        expect: {
+          backstageIdentity: {
+            id: 'daveboyle',
+          },
+          providerInfo: {
+            accessToken:
+              'ajakljsdoiahoawxbrouawucmbawe.awkxjemaneasdxwe.sodijxqeqwexeqwxe',
+            scope: 'read:user',
+            expiresInSeconds: undefined,
+            idToken: undefined,
+          },
+          profile: {
+            displayName: 'Dave Boyle',
+            email: 'daveboyle@gitlab.org',
+          },
+        },
+      },
+    ];
+
+    for (const test of tests) {
+      it(test.testCase, () => {
```

Comment:
I'd like to see these tests be individual test cases so we can see the setup and assertion in the code block rather than refactoring like this. Not saying it's bad, just keeps with the convention of the project

Reply:
Will change. Heads up, the gitlab equivalent test is also being data-driven

Reply:
Perhaps `it.each` would be suitable?

Reply:
Which do you prefer? Separate or more "data-driven"?

```
const add = (a, b) => a + b;

const cases = [[2, 2, 4], [-2, -2, -4], [2, -2, 0]];

describe("'add' utility", () => {
  test.each(cases)(
    "given %p and %p as arguments, returns %p",
    (firstArg, secondArg, expectedResult) => {
      const result = add(firstArg, secondArg);
      expect(result).toEqual(expectedResult);
    }
  );
});

```

Reply:
@benjdlambert  & @freben Added a file for each variation. Pick your flavor and I will drop the other one

Reply:
Personally, I prefer explicit test cases rather than data driven test cases. They're much easier to read rather than arrays of arguments. What do you think @freben?

Reply:
Go for it!

Reply:
All right 💪  Updated